src/org/sosy_lab/java_smt/solvers/bitwuzla/BitwuzlaFormulaManager.java

```diff
@@ -0,0 +1,265 @@
+
+    // After the run, get the final assertion from the parser
+    Vector_Term assertions = parser.bitwuzla().get_assertions();
+    Preconditions.checkArgument(
+        !assertions.isEmpty(), "No assertion found in input string \"%s\"", formulaStr);
+    Term result = Iterables.getLast(assertions);
+
+    // Now get all symbols that were declared in the input
+    Vector_Term declared = parser.get_declared_funs();
+
+    // Process the symbols from the parser
+    Map_TermTerm subst = new Map_TermTerm();
+    for (Term term : declared) {
+      if (cache.containsRow(term.symbol())) {
+        // Symbol is from the context: add the original term to the substitution map
+        subst.put(term, cache.get(term.symbol(), term.sort()));
+      } else {
+        // Symbol is new, add it to the context
+        cache.put(term.symbol(), term.sort(), term);
+      }
+    }
+
+    // Substitute all symbols from the context with their original terms
+    result = creator.getTermManager().substitute_term(result, subst);
+
+    // Return the updated term
+    return creator.encapsulateBoolean(result);
+  }
+
+  private String stripSMTLIB2String(String pFormulaStr) {
+    String s = pFormulaStr;
+    int setLogicIndex = s.indexOf("(set-logic ");
+    if (setLogicIndex != -1) {
+      int endLogicIndex = s.indexOf(')', setLogicIndex + 1);
+      String s1 = s.substring(0, setLogicIndex);
+      String s2 = s.substring(endLogicIndex + 1);
+      s = s1 + s2;
+    }
+    if (s.contains("(check-sat)")) {
+      s = s.replace("(check-sat)", "");
+    }
+    if (s.contains("(exit)")) {
+      s = s.replace("(exit)", "");
+    }
+    return s;
+  }
+
+  @Override
+  public Appender dumpFormula(Term pTerm) {
+    // There are 2 ways of SMT2 printing in Bitwuzla, bitwuzla_term_print() and
+    // bitwuzla_term_print_fmt(), which print a single formula, and bitwuzla_print_formula(),
+    // which prints the complete assertion stack of the bitwuzla instance given to the function.
+    // Only bitwuzla_print_formula() gives us the proper SMT2 format, with (check-sat) etc.
+    // Note: bitwuzla_print_formula() is wrapped in dump_assertions_smt2()
+    return new Appenders.AbstractAppender() {
+      @Override
+      public void appendTo(Appendable out) throws IOException {
+        Bitwuzla bitwuzla = new Bitwuzla(creator.getTermManager());
+        for (Term t : creator.getVariableCasts()) {
+          bitwuzla.assert_formula(t);
+        }
+        bitwuzla.assert_formula(pTerm);
+        String dump = bitwuzla.print_formula();
+        if (dump.startsWith("(set-logic ")) {
+          dump = dump.substring(1 + dump.indexOf(')'));
+        }
+        dump = dump.replace("(check-sat)", "");
+        dump = dump.replace("(exit)", "");
+        out.append(dump);
+      }
+    };
+  }
+
+  static Term getBitwuzlaTerm(Formula pT) {
+    return ((BitwuzlaFormula) pT).getTerm();
+  }
+
+  // Split up a sequence of lisp expressions
```

Comment:
Are we sure we want to do tokenization for SMTLIB on our own?

Does this consider special cases like `(define-const |(some weird name)| Int)` and string constants? :-)

Reply:
No we don't want to do this, but it works for now. I will ask the devs to provide a method for us that takes a arbitrary SMTIB2 String and returns a Term.
If this does not work, i'll add FP to the independent SMTLIB2 layer and add it using that.